feat: API to process multiple ixs with one TransactionContext

Duckaet opened the pull request:
it implemenst the issue #185

buffalojoec commented on harness/src/compile_accounts.rs:
```diff
@@ -63,3 +63,63 @@ where
+pub fn compile_accounts_chain<'a, 'b>(
+    instructions: impl Iterator<Item = &'b Instruction>,
+    accounts: impl Iterator<Item = &'a (Pubkey, Account)>,
+    loader_key: Option<Pubkey>,
```
We need to actually capture all owners of each program in the top-level instruction set, so we'll want to use something like a `Vec<(Pubkey, Pubkey)>` where we pass `(program_id, owner)`.

Then, we can stub only the top-level instructions' program accounts. What you have here won't work for a transaction context with two different program IDs invoked.

buffalojoec reviewed (commented):
Thanks for throwing this together! It's going to be a pretty hefty change - requiring lots of tests - so bear with me, but we can move toward a working solution.

Duckaet commented:
@buffalojoec , i tried to call compile_accounts but it broke some tests.. because they build different KeyMaps for single vs all instruction
```
KeyMap::compile(instruction_index, std::iter::once(instruction))
KeyMap::compile_for_chain(instructions.iter())
```

extracted out functions from process_transaction_instructions and validate

fixed the program owner bug, added more tests to verify the functionality

i have reduced duplicate code aswell with extract func

buffalojoec commented:
@Duckaet I just wrote a PR https://github.com/anza-xyz/mollusk/pull/187 that should make this change a lot easier. I refactored the way account fallbacks are handled during the account compilation step, which should enable you to share a `KeyMap` more easily.
